sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/ReplicatedResourceClient.java

```diff
@@ -158,48 +155,94 @@ public Mono<StoreResponse> invokeAsync(RxDocumentServiceRequest request,
 
     private Mono<StoreResponse> getStoreResponseMonoWithSpeculation(RxDocumentServiceRequest request, Quadruple<Boolean, Boolean, Duration, Integer> forceRefreshAndTimeout) {
         CosmosEndToEndOperationLatencyPolicyConfig config = request.requestContext.getEndToEndOperationLatencyPolicyConfig();
+        List<String> preferredRegions = this.transportClient.getGlobalEndpointManager().getPreferredRegions();
+        AvailabilityStrategy strategy = config.getAvailabilityStrategy();
+        if (strategy == null){
+            strategy = new DefaultAvailabilityStrategy();
+        }
+        if (strategy instanceof ThresholdBasedAvailabilityStrategy) {
+                speculativeProcessor = new ThresholdBasedSpeculation();
+        }
+
         List<Mono<StoreResponse>> monoList = new ArrayList<>();
         List<RxDocumentServiceRequest> requests = new ArrayList<>();
 
-        if (speculativeProcessor.shouldIncludeOriginalRequestRegion()) {
+        if (speculativeProcessor != null) {
+            List<String> effectiveRetryRegions = strategy.getEffectiveRetryRegions(preferredRegions,
+                request.requestContext.getExcludeRegions());
+            if (effectiveRetryRegions.size() > BridgeInternal.getNumberOfRegionsToTry(strategy)){
+                effectiveRetryRegions = effectiveRetryRegions.subList(0, BridgeInternal.getNumberOfRegionsToTry(strategy));
+            }
+            effectiveRetryRegions
+                .forEach(regionName -> {
+                    URI locationURI = getLocationURIFromAvailableEndPoints(regionName, request);
+                    if (locationURI != null) {
+                        RxDocumentServiceRequest newRequest = request.clone();
+                        newRequest.requestContext.routeToLocation(locationURI);
+                        requests.add(newRequest);
+                        monoList.add(getStoreResponseMono(newRequest, forceRefreshAndTimeout)
+                            .delaySubscription(speculativeProcessor.getThreshold(config)
```

Comment:
even for the first region it is going to be delayed?

Reply:
This was stale. Fixed it, thanks for catching